### ISSUE TITLE
Cannot run hello world on Ubuntu

### ISSUE THREAD

**Issue by renatoathaydes** [Concurnas/Concurnas#11]
I have created the following file:

**main.conc**
```java
System.out.println("hello world")
```

Compiled with:

```
$ bin/concc main.conc
```

This created `main.class` in the local dir. All good.

Trying to run with `conc` results in:

```
➜  Concurnas-1.14.015 bin/conc main     
java.lang.SecurityException: Prohibited package name: java.lang
	at java.base/java.lang.Throwable.init(Throwable.java:270)
	at java.base/java.lang.Exception.init(Exception.java:66)
	at java.base/java.lang.RuntimeException.init(RuntimeException.java:62)
	at java.base/java.lang.SecurityException.init(SecurityException.java:52)
	at java.base/java.lang.ClassLoader.preDefineClass(ClassLoader.java:898)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1014)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:877)
	at com.concurnas.runtime.ConcurnasClassLoader.defineClassSupportingEtc(ConcurnasClassLoader.java:268)
	at com.concurnas.runtime.ConcurnasClassLoader.defineClass(ConcurnasClassLoader.java:218)
	at com.concurnas.runtime.ConcurnasClassLoader.loadClass(ConcurnasClassLoader.java:451)
	at com.concurnas.runtime.ConcurnasClassLoader.loadClass(ConcurnasClassLoader.java:390)
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3172)
	at java.base/java.lang.Class.privateGetPublicMethods(Class.java:3197)
	at java.base/java.lang.Class.getMethods(Class.java:1910)
	at com.concurnas.conc.Conc.getMethod(Conc.java:324)
	at com.concurnas.conc.Conc.doMainLoop(Conc.java:253)
	at com.concurnas.conc.Conc.doit(Conc.java:180)
	at com.concurnas.conc.Conc.main(Conc.java:72)
Unknown error in conc: Failure during definition of class: java.lang.String
java.lang.RuntimeException: Failure during definition of class: java.lang.String
	at com.concurnas.runtime.ConcurnasClassLoader.defineClass(ConcurnasClassLoader.java:221)
	at com.concurnas.runtime.ConcurnasClassLoader.loadClass(ConcurnasClassLoader.java:451)
	at com.concurnas.runtime.ConcurnasClassLoader.loadClass(ConcurnasClassLoader.java:390)
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3172)
	at java.base/java.lang.Class.privateGetPublicMethods(Class.java:3197)
	at java.base/java.lang.Class.getMethods(Class.java:1910)
	at com.concurnas.conc.Conc.getMethod(Conc.java:324)
	at com.concurnas.conc.Conc.doMainLoop(Conc.java:253)
	at com.concurnas.conc.Conc.doit(Conc.java:180)
	at com.concurnas.conc.Conc.main(Conc.java:72)
Caused by: java.lang.SecurityException: Prohibited package name: java.lang
	at java.base/java.lang.Throwable.init(Throwable.java:270)
	at java.base/java.lang.Exception.init(Exception.java:66)
	at java.base/java.lang.RuntimeException.init(RuntimeException.java:62)
	at java.base/java.lang.SecurityException.init(SecurityException.java:52)
	at java.base/java.lang.ClassLoader.preDefineClass(ClassLoader.java:898)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1014)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:877)
	at com.concurnas.runtime.ConcurnasClassLoader.defineClassSupportingEtc(ConcurnasClassLoader.java:268)
	at com.concurnas.runtime.ConcurnasClassLoader.defineClass(ConcurnasClassLoader.java:218)
	... 10 more
```

**Comment by renatoathaydes** [comment]
Forgot to mention, java version:

```
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)
```

**Comment by jasontatton** [comment]
Thanks for raising this. This is an interesting problem, it seems that attempting to execute a compiled class which is defined in a parent directory to the location of the conc script results in the main classloader for Concurnas failing to initialize properly. 

Say our installation path for Concurnas is ~/collstuff/concurnas/ - then executing code which resides in ~/collstuff/concurnas/, ~/collstuff/ or ~/ will result in an error. Note however that if the code is placed in a non direct parent directory, e.g. ~/collstuff/myProject/<code/classes in here> then this works fine.

So for now a workaround is to place the code in a location which isn't a parent directory

**Comment by jasontatton** [comment]
This bug is now resolved in Concurnas version 1.14.017 onwards.
